pyannote/audio/core/model.py

```diff
@@ -468,9 +477,8 @@ def __by_name(
         if isinstance(modules, str):
             modules = [modules]
 
-        for name, module in ModelSummary(self, max_depth=-1).named_modules:
-            if name not in modules:
-                continue
+        for name in modules:
+            module = getattr(self, name)
```

Comment:
What happens if no module with this `name` exists?

Reply:
This will raise an `AttributeError: <Model> object has no attribute <name>`.

Reply:
we could raise an exception telling the user that there is no module named `name` for this model